sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -361,11 +379,38 @@ object ViewHelper {
     }
   }
 
+  /**
+   * Convert the view query SQL configs in `properties`.
+   */
+  private def generateQuerySQLConfigs(conf: SQLConf): Map[String, String] = {
+    val modifiedConfs = conf.getAllConfs.filter { case (k, _) =>
+      conf.isModifiable(k) && !isConfigBlacklisted(k)
+    }
+    val props = new mutable.HashMap[String, String]
+    if (modifiedConfs.nonEmpty) {
+      val confJson = compact(render(JsonProtocol.mapToJson(modifiedConfs)))
+      props.put(VIEW_QUERY_SQL_CONFIGS, confJson)
```

Comment:
have you stress-tested this? The hive metastore has a limitation about property value length. You can take a look at `HiveExternalCatalog.tableMetaToTableProps`.

Another idea is to put one config per table property entry.

Reply:
Thanks for pointing this out. I changed to store one config per table property entry, each with a "view.sqlConfig." prefix.